src/modules/paymentProcessor/PP_Streaming_v1.sol

```diff
@@ -915,4 +949,79 @@ contract PP_Streaming_v1 is Module_v1, IPP_Streaming_v1 {
         );
         return (success && data.length != 0 && _token.code.length != 0);
     }
+
+    function _validOriginAndTargetChain(
+        uint originChainId_,
+        uint targetChainId_
+    ) internal view returns (bool) {
+        return (
+            (originChainId_ == targetChainId_)
+                && (originChainId_ == block.chainid)
+        );
+    }
+
+    function _getStreamingDetails(
+        bytes32 orderFlags_,
+        bytes32[] memory orderData_
+    ) internal view returns (uint start_, uint cliff_, uint end_) {
+        uint[3] memory returnData;
+
+        uint8 positionInOrderData = 0;
+        uint8 positionInReturnData = 0;
+
+        for (uint i = 0; i < 256; i++) {
+            if (positionInReturnData == returnData.length) {
+                // we have either:
+                // - reached the end of the orderData_ array
+                // - already checked for all the values this P_P will need
+                //      ==> exit loop
+                break;
+            }
+
+            bool orderBit = (uint(orderFlags_) & (1 << i)) != 0;
+            bool processorBit = (uint(PROCESSOR_FLAGS) & (1 << i)) != 0;
+
+            if (orderBit == true && processorBit == false) {
+                // the P_P does not use that value
+                //      ==> skip that data slot in the order
+                positionInOrderData++;
+            }
+            if (orderBit == false && processorBit == true) {
+                // the P_P needs the value, but it's missing in the order
+                //      ==> use default value
```

Comment:
Would it not be better (safer) if this is an error? This after all is not a case where the order does not provide a value, it is a case where the "order type" is different from what is expected (I mean that if the flag value is different, this is not the same as providing a "0" value and have the PP fill that in with a default value)
For example, if the order says "This is an order with no start and end date", then it is it is better that a PP that is made for streaming orders says "Ok, you probably made a istake sending me this order" rather than have it say "O, ok, then I will just make it a streaming order which starts in 17 days and lasts 189 hours"